tests/Unleash.Tests/ClientFactory/SyncStartupUnitTest.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+using Unleash.Communication;
+using Unleash.Metrics;
+using FakeItEasy;
+using Unleash.ClientFactory;
+using Unleash.Serialization;
+using System.Threading.Tasks;
+
+namespace Unleash.Tests.StartupBehavior
+{
+    public class SyncStartupUnitTest
+    {
+
+        [Test]
+        public async Task AsyncStartupClientStartup()
+        {
+            var mockApiClient = A.Fake<IUnleashApiClient>();
+            var settings = new MockedUnleashSettings();
+            settings.UnleashApiClient = mockApiClient;
+            var unleashFactory = new UnleashClientFactory(settings);
+
+            IUnleash unleashTask = await unleashFactory.Generate(SynchronousInitialization: true);
+
+            A.CallTo(() => mockApiClient.FetchToggles(string.Empty, A<CancellationToken>.Ignored)).MustHaveHappenedOnceExactly();
```

Comment:
How do I test that this call was awaited? This only asserts that it was called.

Reply:
Return some fake toggles and assert the expectations of use, like, unleash should have a flag x.

Besides that, only asserting calls should do the trick.

I think that some tests are missing, like:
- What if it fails or throws exceptions?
- What if `SynchronousInitialization` is set to false? uneleash should have or not toggle x?

Reply:
Added tests showing exceptions bubbling, and `SynchronousInitialization`=false